Assets/Scripts/GameplayObjects/GameplayObjectPool.cs

```diff
@@ -0,0 +1,94 @@
+﻿using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+
+public class GameplayObjectPool : IObjectPool
```

Comment:
This class seems to have a "manager" quality - it looks like it's generic, but in fact it deals with very specific objects, each in it's own individual name (look at the 2nd comment here as a good explanation why it's bad: https://softwareengineering.stackexchange.com/questions/129537/can-manager-classes-be-a-sign-of-bad-architecture)
If it creates objects, it should be a factory. if it's a factory, it should create only one type of object.